### PR TITLE
Adding file processing support

### PR DESCRIPTION
**Pull request by dleblanc**
This adds support for File Processing to citrine-python.

Currently the only implementation is 'Process CSV Pre-Ingest".

### PR Type:
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

### Adherence to team decisions
- [x] I have added tests for 100% coverage
- [x] I have written Numpy-style docstrings for every method and class.
- [ ] I have communicated the downstream consequences of the PR to others.
- [x] I have bumped the version in setup.py

### CODE REVIEW

File: src/citrine/resources/file_link.py [hunk 1]
```diff
@@ -236,10 +280,12 @@ def _make_upload_request(self, file_path: str, dest_name: str):
         try:
 
             uploader.region_name = upload_request['s3_region']
-            uploader.aws_access_key_id = upload_request['temporary_credentials']['access_key_id']
+            uploader.aws_access_key_id = upload_request['temporary_credentials'][
+                'access_key_id']
```
Comment (kroenlein) [inline src/citrine/resources/file_link.py:284]: linting clean-up?  There are a couple changes like this that don't really improve legibility.

**Review by kroenlein — APPROVED** [review]
Straight forward, though there are a few odd reformatting choices.

File: src/citrine/resources/file_link.py (hunk 1 above)
Comment (dleblanc) [reply src/citrine/resources/file_link.py:284]: I assume that's my IDE doing its thing. I'll see about backing out some of the useless ones.

**Review by kroenlein — APPROVED** [review]
LGTM

**Review by kroenlein — APPROVED** [review]
LGTM

**Pull request merged**